packages/arb-avm-cpp/data_storage/src/messageentry.cpp

```diff
@@ -37,33 +37,44 @@ MessageEntry deserializeMessageEntry(
     std::vector<const unsigned char> entry_vector) {
     auto current_iter = entry_vector.begin();
 
-    auto message_hash = extractUint256(current_iter);
     auto inbox_hash = extractUint256(current_iter);
     auto block_height = extractUint64(current_iter);
     auto last_message_in_block = current_iter[0] == 1;
+    auto message_size = extractUint64(current_iter);
+    uint64_t remaining_size = entry_vector.end() - current_iter;
+    if (remaining_size < message_size) {
+        message_size = remaining_size;
+    }
```

Comment:
I'm wondering whether we need to store the size, or if we can just assume it's the remaining data in the `entry_vector`